tasks/request_files/request_files.py

```diff
@@ -235,263 +245,213 @@ def inner_task(event: Dict, max_retries: int, retry_sleep_secs: float,
-                    a_file[FILE_SUCCESS_KEY] = True
-                    a_file[FILE_ERROR_MESSAGE_KEY] = ''
-
-                    post_status_for_file_to_queue(
-                        job_id, granule_id, os.path.basename(a_file[FILE_KEY_KEY]),
+                    restore_object(
+                        s3,
                         a_file[FILE_KEY_KEY],
-                        a_file[FILE_DEST_BUCKET_KEY],
-                        ORCA_STATUS_PENDING,
-                        None,
-                        request_time,
-                        datetime.now(timezone.utc).isoformat(),
-                        None,
-                        RequestMethod.POST,
-                        db_queue_url,
-                        max_retries,
-                        retry_sleep_secs)
-
+                        restore_expire_days,
+                        glacier_bucket,
+                        attempt,
+                        job_id,
+                        retrieval_type,
+                    )
+                    a_file[FILE_SUCCESS_KEY] = True
+                    a_file[FILE_ERROR_MESSAGE_KEY] = ""
                 except ClientError as err:
                     LOGGER.warning(err)
                     a_file[FILE_ERROR_MESSAGE_KEY] = str(err)
 
         attempt = attempt + 1
-        if attempt <= max_retries + 1:  # Only sleep if not on last attempt. # todo: Use backoff code. ORCA-201
+        if (
+            attempt <= max_retries + 1
+        ):  # Only sleep if not on last attempt. # todo: Use backoff code. ORCA-201
             # Check for early completion.
-            if all(a_file[FILE_SUCCESS_KEY] for a_file in granule[GRANULE_RECOVER_FILES_KEY]):
+            if all(
+                a_file[FILE_SUCCESS_KEY]
+                for a_file in granule[GRANULE_RECOVER_FILES_KEY]
+            ):
                 break
             time.sleep(retry_sleep_secs)
 
+    files = []
     any_error = False
     for a_file in granule[GRANULE_RECOVER_FILES_KEY]:
         # if any file failed, the whole granule will fail
         if not a_file[FILE_SUCCESS_KEY]:
             any_error = True
-            # If this is reached, that means there is no entry in the db for file's status.
-            post_status_for_file_to_queue(
-                job_id, granule_id, os.path.basename(a_file[FILE_KEY_KEY]),
-                a_file[FILE_KEY_KEY],
-                a_file[FILE_DEST_BUCKET_KEY],
-                ORCA_STATUS_FAILED,
-                a_file.get(FILE_ERROR_MESSAGE_KEY, None),
-                request_time,
-                datetime.now(timezone.utc).isoformat(),
-                None,
-                RequestMethod.POST,
-                db_queue_url,
-                max_retries,
-                retry_sleep_secs)
+            files["status_id"] = shared_recovery.OrcaStatus.FAILED.value
+            files["error_message"] = a_file[FILE_ERROR_MESSAGE_KEY]
+            files["completion_time"] = datetime.now(timezone.utc).isoformat()
```

Comment:
This won't work, since it tries to set dictionary values on an array. I recommend you delete these three and use something like https://stackoverflow.com/questions/8116127/python-conditional-variable-setting to set status_id and completion_time in files.append. For error_message you can do "error_message": files.get("error_message", None)

Reply:
Keep in mind that error_message and completion_time may not be set.